src/Microsoft.OpenApi.OData.Reader/Operation/OperationHandler.cs

```diff
@@ -289,5 +296,23 @@ protected virtual void SetCustomLinkRelType()
                 }
             }
         }
+
+        /// <summary>
+        /// Set string representation of the Edm Target Path for annotations
+        /// </summary>
+        protected virtual void SetTargetPath()
+        {
+            var targetPath = new StringBuilder(Context.Model.EntityContainer.FullName());
+
+            bool skipLastSegment = Path.LastSegment is ODataRefSegment;
+            foreach (var segment in Path.Segments.Where(segment => segment is not ODataKeySegment 
+                && segment is not ODataTypeCastSegment // TODO: Presence of type casts in the TargetPath is currently throwing exceptions in Edm lib. Remove after update.
```

Comment:
Out of curiosity, do we know why this errors are occurring? Could we raise an issue with the library?

Reply:
I created [this](https://github.com/OData/odata.net/issues/2921) issue in the library repo. It has more details on what's causing the error.

Reply:
Thanks for this. Looks like a PR is already underway to fix it. https://github.com/OData/odata.net/pull/2922/files

Could we create an issue on this Repo as well to add in the change when a new version of the library is released to undo this change?

Reply:
The issue was created here https://github.com/microsoft/OpenAPI.NET.OData/issues/515